smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpBindingProtocolGenerator.kt

```diff
@@ -789,22 +576,28 @@ abstract class HttpBindingProtocolGenerator : ProtocolGenerator {
                 // payload member(s)
                 val httpPayload = responseBindings.firstOrNull { it.location == HttpBinding.Location.PAYLOAD }
                 if (httpPayload != null) {
-                    renderExplicitHttpPayloadDeserializer(ctx, httpPayload, bodyDeserializerName, writer)
+                    renderExplicitHttpPayloadDeserializer(ctx, httpPayload, writer)
                 } else {
                     // Unbound document members that should be deserialized from the document format for the protocol.
-                    // The generated code is the same across protocols and the serialization provider instance
-                    // passed into the function is expected to handle the formatting required by the protocol
                     val documentMembers = responseBindings
                         .filter { it.location == HttpBinding.Location.DOCUMENT }
                         .sortedBy { it.memberName }
                         .map { it.member }
 
                     if (documentMembers.isNotEmpty()) {
-                        writer
-                            .addImport(RuntimeTypes.Http.readAll)
-                            .write("val payload = response.body.#T()", RuntimeTypes.Http.readAll)
+                        val sdg = structuredDataParser(ctx)
```

Comment:
structured data generator

Reply:
Ah, I see that now. It's confusing because the function's named `structuredDataParser` but it returns a generator and your local variable's name ignores the "parsing" part. I might suggest `sdpg` or perhaps just `generator` for the variable name.